packages/story-editor/src/components/library/panes/pageTemplates/savedTemplates.js

```diff
@@ -135,6 +135,9 @@ function SavedTemplates({ pageSize, loadTemplates, isLoading, ...rest }) {
   });
 
   // TODO: Maybe use proper reducer here like in useContextValueProvider.
+  // I think we'll need to take some measures with our approach here
+  // to not bog down the users machine. Possibly use the Background
+  // task api like the carousel and do them sequentially?
```

Comment:
This sounds like a smart idea. Prometheus has done substantial work on this in the carousel and it would be great (like you said) to even have a util that all of these cases can use - could probably reuse that work some or at least let it guide the implementation. 

Curious what @miina and @swissspidy think.

Reply:
Thanks for tagging! Assuming that a newly saved template will have an image generated while creating it, and this would mainly be a question for when backfilling all the previously existing templates? Makes sense to me -- agreed that we should try to do it so that the normal usage of the editor wouldn't be disturbed / start lagging unexpectedly.
Tagging @barklund here as well since he implemented the carousel part of image generation.

Reply:
yea I think this was initially created for backfilling previously existing ones, but it's pretty self contained and implementation agnostic, so theoretically, if we're cool with it, we can just use it anywhere.

I agree that if we've already generated a blob, we should pass that around instead of continually generating one cus it is an expensive task. Might have to hoist up some of the thumbnail blob state to be able to pass it around. 

Anyway, I think this provides us a flexible way to generate blobs without being tied to the VDOM structure (Don't have to worry about managing refs, size of the page node in the UI that we're generating an image from etc). 

The next steps would be minimizing the generation of blobs & taking measures to spread out blob generation to not bog down the users machine.

Mainly wanted to get feedback on the page data -> jpeg function I set up here cus that will have implications on whatever approach we take in the coming steps

Reply:
@swissspidy - thoughts on this?

Reply:
I like the simplicity of that `storyPageToBlob` function you created there!

I think this will be very useful for the planned enhancement to use these images in more places. @barklund just created a ticket for this moments ago which aligns very well with your ideas: #9749

That ticket is probably the best place to talk about caching, background tasks, etc.

As for saved templates specifically, the backfilling done here only happens once and only when this panel is open. Afterwards we always display the existing image. So we don't need to optimize this use case too much. The carousel and grid view are more important.

Reply:
awesome! yea that's a good point about this only ever happening once. noted.